libfs/fs.go

```diff
@@ -366,15 +387,22 @@ func (fs *FS) Join(elem ...string) string {
 
 // TempFile implements the billy.Filesystem interface for FS.
 func (fs *FS) TempFile(dir, prefix string) (billy.File, error) {
-	// We'd have to turn off journaling to support TempFile fully.
-	return nil, errors.New("TempFile is not currently supported")
+	// We'd have to turn off journaling to support TempFile perfectly,
+	// but the given uniq ID and a random number should be good
+	// enough.  Especially since most users will end up renaming the
+	// temp file before journal flushing even happens.
+	r := rand.Int63()
```

Comment:
8 bytes seems like plenty for the purposes of making unique filenames within a KBFS folder, but I'm open to suggestions.  I guess longer file names will take slightly longer to process internally, but it probably wouldn't affect performance much.  What do you suggest for an appropriate number of bytes?